src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -193,56 +200,66 @@ function MultiSelect<T>({
         setSelectedItems((items) => items.filter((item) => itemId(item) !== matcher));
     };
 
-    
+
 
     /**
      * selects and deselects an item from selection list
      * if the item exists it removes it instead
      * @param item
      */
     const onItemSelect = (item: T) => {
-    
+
         if (itemHasBeenSelectedAlready(itemId(item))) {
             removeItemSelection(itemId(item));
         } else {
             setSelectedItems((items) => [...items, item]);
         }
-        
+
         //remove if already exist
         if (createdSelectedItems.find((t) => itemLabel(t) === itemLabel(item))) {
            setCreatedSelectedItems((prevItems) => prevItems.filter(prevItem => itemLabel(prevItem) !== itemLabel(item)))
         }else {
             const wasNewlyCreated = createdItems.find((t) => itemLabel(t) === itemLabel(item));
-            //only add to createdSelectedItems if it was previously created and not 
-            // from the initial items or a possible query response 
+            //only add to createdSelectedItems if it was previously created and not
+            // from the initial items or a possible query response
             if(wasNewlyCreated){
                  setCreatedSelectedItems((prevItems) =>[...prevItems, item])
             }
         }
- 
+
         inputRef.current?.select();
     };
 
     /**
      * search through item list using "label prop" and update the items popover
      * @param query
      */
-    const onQueryChange = async (query: string) => {
-        if (query.length) {
-            setQuery(query);
-            setFilteredItemList([])
-            const resultFromQuery = runOnQueryChange && (await runOnQueryChange(removeExtraSpaces(query)));
-            setFilteredItemList(() =>
-                [...(resultFromQuery ?? itemsCopy), ...createdItems].filter(item =>
-                    itemLabel(item).toLowerCase().includes(query.toLowerCase())
-                )
-            );
+    const onQueryChange = (query: string) => {
+        if (query.length && query !== requestState.current.query) {
+            requestState.current.query = query
+            if(requestState.current.timeoutId) {
+                clearTimeout(requestState.current.timeoutId)
+            }
+            const fn = async () => {
+                // TODO: Show spinner instead
```

Comment:
What about this TODO, "show spinner" looks like it would be important as feedback for the user?